prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -136,16 +137,36 @@ class EmailScheduleEngine {
 
             // get the orgs to fire for
             orgs.forEach { org ->
+                val countOfRecords = workflowEngine.db.fetchDownloadableReportFiles(
+                    OffsetDateTime.now().minusDays(1L),
+                    org
+                ).size;     
+                val template = getTemplate(schedule.template, countOfRecords );     
                 val emails: Iterable<String> = if (schedule.emails.size > 0) schedule.emails else getEmails(org, logger)
-                logger.info("EmailEngineFunction:: processing organization $org within ${schedule.template}")
+                logger.info("EmailEngineFunction:: processing organization $org within ${template}")
                 emails.forEach { email ->
-                    logger.info("EmailEngineFunction:: sending email to $email")
-                    dispatchToSendGrid(schedule.template, listOf(email), logger)
+                    logger.info("EmailEngineFunction:: sending email template ${template} to $email")
+                    dispatchToSendGrid(template, listOf(email), logger)
                 }
             }
         }
     }
 
+    /**
+     * Get the template to use
+     * 
+     * @param template the template string from the database
+     * @param count of records found in the last 24hrs
+     * 
+     * @returns template name to use
+     */
+    private fun getTemplate( template: String, count: Int ): String {
+        var retValue = template;
+        if( template.split(":").size > 1 )
```

Comment:
question rather than a request for change: what happens when the template contains more than 1 `:` and thus `.size` return `(2,...[` ? Is that possible in the future? If so, I would prefer a(n) warning/error/critical logging or something that tells us about this.